example/torus.red

```diff
@@ -15,58 +15,40 @@ data torus where
   ]
 
 let t2c : torus → s1 × s1 =
-  λ [
+  elim [
   | pt → (base, base)
   | p/one i → (loop i, base)
   | p/two i → (base, loop i)
   | square i j → (loop j, loop i)
   ]
 
 let c2t/base : s1 → torus =
-  λ [
+  elim [
   | base → pt
   | loop i → p/two i
   ]
 
-let c2t/transpose : s1 → s1 → torus =
-  λ [
+let c2t : (s1 × s1) → torus =
+  λ [,] →
```

Comment:
Using this inversion gadget, we avoid the two-step definition that we had before.